openvdb_points/tools/AttributeSet.h

```diff
@@ -305,6 +337,7 @@ class AttributeSet::Descriptor
     size_t insert(const std::string& name, const NamePair& typeName);
     NameToPosMap                mNameMap;
     std::vector<NamePair>       mTypes;
+    NameToPosMap                mGroupMap;
```

Comment:
A group attribute array is essentially a char or uint8 (an 8-bit value). Each bit can be assigned to a different group, hence in order to have more than 8 groups, you'd need more than one attribute array.

If you were to strip out all group attribute arrays from the set and put them into a line, the offset is the index associated with the bit you care about, or the relative "offset" into the group attribute arrays.

The key advantage of this approach is that adding, dropping or re-ordering non-group attribute arrays from the set has no impact on how groups are interpreted and simply uses the underlying attribute set for the storage.

I'll admit this can be a little obscure at first, I'll add some more documentation to properly explain this.